modules/i3s/src/lib/parsers/parse-slpk/slpk-archieve.ts

```diff
@@ -130,7 +143,12 @@ export class SLPKArchive {
    * @returns buffer with the file data
    */
   private async getDataByPath(path: string): Promise<ArrayBuffer | undefined> {
-    const data = await this.getFileBytes(path);
+    // sometimes paths are not in lower case when hash file is created,
+    // so first we're looking for original file name and then for lower case
+    let data = await this.getFileBytes(path);
+    if (!data) {
+      data = await this.getFileBytes(path.toLocaleLowerCase());
+    }
```

Comment:
Datasets made by spec should work faster. As `lowercase` is by spec so it should work faster. I'd suggest to seek it first. If search by spec is not working the code will try to find it in another way.
Point it out in the comment with reference to the spec.